packages/atomic/src/components/result-lists/atomic-result-children/atomic-result-children.tsx

```diff
@@ -140,31 +162,91 @@ export class AtomicResultChildren {
     );
   }
 
-  public render() {
-    if (!this.ready) return null;
-    if (this.templateHasError) return <slot></slot>;
+  private showCollapseButton() {
     const result = this.getResult();
-    if (result.children.length) {
-      return (
-        <Host>
-          <slot name="before-children"></slot>
-          {result.isLoadingMoreResults ? (
-            <Loading></Loading>
-          ) : (
-            result.children.map((child) => this.renderChild(child))
-          )}
-          <slot name="after-children"></slot>
-        </Host>
-      );
+    return (
+      Boolean(result.children.length) &&
+      !result.isLoadingMoreResults &&
+      !result.moreResultsAvailable
+    );
+  }
+
+  private showChildrenWrapper() {
+    const result = this.getResult();
+    if (!this.initialChildren.length && this.hideResults) return false;
+    return (
+      Boolean(result.children.length) ||
+      result.isLoadingMoreResults ||
+      this.showAllResults
+    );
+  }
+
+  public componentWillUpdate() {
+    if (this.getResult().children && !this.initialChildren) {
+      this.initialChildren = this.getResult().children;
+    }
+    if (this.getResult().isLoadingMoreResults && !this.showAllResults) {
+      this.showAllResults = true;
+    }
+  }
+
+  private getComponents() {
+    const components = [];
+    const result = this.getResult();
+    const hasResults = Boolean(result.children.length);
+    if (hasResults) {
+      components.push(<slot name="before-children"></slot>);
     }
     if (result.isLoadingMoreResults) {
-      return <Loading></Loading>;
+      components.push(
+        <ListDisplayResultsPlaceholder
+          classes="child-result"
+          resultsPerPageState={{
+            numberOfResults: result.children.length || 2,
+          }}
+          density={this.displayConfig.density}
+          imageSize={this.imageSize || this.displayConfig.imageSize}
+        />
+      );
+    } else if (hasResults) {
+      const children = this.hideResults
+        ? this.initialChildren
+        : result.children;
+      components.push(children.map((child) => this.renderChild(child)));
+    } else if (this.showAllResults) {
+      components.push(
+        <p part="no-result-root" class="no-result-root">
+          {this.noResultText || this.bindings.i18n.t('no-documents-related')}
+        </p>
+      );
     }
-    return null;
+    if (hasResults) {
+      components.push(<slot name="after-children"></slot>);
+    }
+    return components;
   }
-}
 
-const Loading: FunctionalComponent = () => {
-  // TODO:
-  return <p>Loading...</p>;
-};
+  public render() {
+    if (!this.ready) return null;
+    if (this.templateHasError) return <slot></slot>;
+    return (
+      <Host>
+        {this.showCollapseButton() && (
```

Comment:
Instead of having extracting the condition and rendering directly in `render`, I would prefer to extract the whole thing and follow a pattern like this:
```tsx
private renderCollapseButton() {
  if (!someConditions) {
    return;
  }
  return <Button ...>...</Button>;
}

public render() {
  ...
  return (
    <Host>
      {this.renderCollapseButton()}
    </Host>
  );
}
```

Reply:
okie